### PR TITLE
More Element Tests

### LINKED ISSUE #1349: Unit Tests on Elements
Add a method to every element to inverse/reverse the element. Negate `ds` (thick) or negate the _strength_ (thin), then all `compute_constants()`.

## Corner Cases

- `PRot`: swap `phi_in` ↔ `phi_out`
- `DipEdge`: negate `psi / K` coefficients, swap `entry`↔`exit`
- `RFCavity` (thick, energy-changing) needs `-ds` plus negating escale or shifting phase by 180° to reverse the acceleration
- Apertures: nothing for transmitted particles, not inversible for absorbed particles

### PR DESCRIPTION
**Pull request by ax3l**
Test our elements more systematically.

Close #1349

### CODE REVIEW

File: src/elements/Drift.H [hunk 1]
```diff
@@ -74,6 +74,9 @@ namespace impactx::elements
         {
         }
 
+        /** Invert the element (negate ds) */
+        void invert () { Thick::invert(); }
```
Comment (ax3l) [inline src/elements/Drift.H]: I am not yet sure if I like this as in-place method or in practice returning a fresh, inverted copy is better. I tend to change it to the latter for practical use cases. But then, we can always implement `.copy()` too and do `.copy().invert()`

File: src/elements/RFCavity.H [hunk 2]
```diff
@@ -363,7 +366,7 @@ namespace impactx::elements
 
             // specify constants
             using ablastr::constant::math::pi;
-            amrex::ParticleReal const zlen = m_ds;
+            amrex::ParticleReal const zlen = std::abs(m_ds);
```
Comment (ax3l) [inline src/elements/RFCavity.H:373]: This is the fix I pushed into #1397

File: src/elements/SoftQuad.H [hunk 3]
```diff
@@ -362,7 +365,7 @@ namespace impactx::elements
 
             // specify constants
             using ablastr::constant::math::pi;
-            amrex::ParticleReal const zlen = m_ds;
+            amrex::ParticleReal const zlen = std::abs(m_ds);
```
Comment (ax3l) [inline src/elements/SoftQuad.H:372]: See #1397

File: src/elements/SoftSol.H [hunk 4]
```diff
@@ -371,7 +374,7 @@ namespace impactx::elements
 
             // specify constants
             using ablastr::constant::math::pi;
-            amrex::ParticleReal const zlen = m_ds;
+            amrex::ParticleReal const zlen = std::abs(m_ds);
```
Comment (ax3l) [inline src/elements/SoftSol.H:381]: See #1397

File: tests/python/test_reversibility_elements.py [hunk 5]
```diff
@@ -0,0 +1,476 @@
...
+    cols = PHASE_COLS + (SPIN_COLS if spin else [])
+    any_changed = False
+    for c in cols:
+        if not np.allclose(df[c].to_numpy(), initial[c], atol=1e-14, rtol=0):
+            any_changed = True
+            break
+    assert any_changed, "Forward push did not change any particle coordinate"
+
+
+def check_roundtrip(pc, initial, atol=1e-10, spin=False):
```
Comment (ax3l) [inline tests/python/test_reversibility_elements.py]: We gotta check the expected tolerance we want (absolute/relative). The current atol is just a draft and seems to need a larger value for w/ spin. But not sure if atol w/o an rtol is sensible at all here.

File: src/elements/LinearMap.H [hunk 6]
```diff
@@ -97,6 +97,31 @@ namespace impactx::elements
             return max_err <= tol;
         }
 
+        /** Invert the element (compute symplectic inverse of transport map)
+         *
+         * For a symplectic matrix R, the inverse is
+         *   R^{-1} = Omega^T R^T Omega = -Omega R^T Omega
+         * where Omega is the standard 6x6 symplectic form.
```
Comment (ax3l) [inline src/elements/LinearMap.H]: Renamed in the other PR to:
```suggestion
         *   R^{-1} = J^T R^T J = -J R^T J
         * where J is the standard 6x6 symplectic form.
```

**Review by cemitch99 — APPROVED** [review]
This looks great to me, provided tests pass.  I will try to understand the SpinMap case offline.  It's ok with me to merge this one and continue ongoing testing of special cases.
